plugins/lua/ts_lua.c

```diff
@@ -469,12 +495,18 @@ TSPluginInit(int argc, const char *argv[])
 
   int states                           = TS_LUA_MAX_STATE_COUNT;
   int reload                           = 0;
+  TSMgmtInt mgmt_states                = 0;
   static const struct option longopt[] = {
     {"states", required_argument, 0, 's'},
     {"enable-reload", no_argument, 0, 'r'},
     {0, 0, 0, 0},
   };
 
+  if (TS_SUCCESS == TSMgmtIntGet(ts_lua_mgmt_state_str, &mgmt_states) && 0 < mgmt_states) {
```

Comment:
one more condition should be that mgmt_states cannot be larger than TS_LUA_MAX_STATE_COUNT

Reply:
I think we have checks afterwards to make sure of that. So it should be fine